pyaedt/modeler/Primitives3D.py

```diff
@@ -786,3 +790,316 @@ def get_3d_component_object_list(self, componentname):
+            }
+
+        Parameters
+        ----------
+        actor_folder: str
+            Path to the actor folder. It must contain a json settings file and a 3dcomponent (.a3dcomp)
+        speed:  float, Optional
+            Object movement speed with time
+        global_offset: list, Optional
+            Offset from Global Coordinate System.
+        yaw: float, Optional
+            Yaw Rotation from Global Coordinate System.
+        pitch: float, Optional
+            Pitch Rotation from Global Coordinate System.
+        roll: float, Optional
+            Roll Rotation from Global Coordinate System.
+        flapping_rate: float, Optional
+            Motion Flapping Rate in Hz.
+        Returns
+        -------
+        :class: `pyaedt.modeler.MultiPartComponent.Bird`
+
+        Examples
+        --------
+        >>> from pyaedt import Hfss
+        >>> app = Hfss()
+        >>> bird_folder = "path/to/bird/folder"
+        >>> bird1 = app.modeler.primitives.add_bird(bird_folder, 1.0, [19, 4, 3], 120,-5, flapping_rate=30)
+
+        """
+        self._initialize_multipart()
+
+        if not self._check_actor_folder(actor_folder):
+            return False
+        bird = Bird(actor_folder, speed=speed, flapping_rate=self._arg_with_dim(flapping_rate,"Hz"))
+        bird.offset = global_offset
+        bird.yaw = self._arg_with_dim(yaw, "deg")
+        bird.pitch = self._arg_with_dim(pitch, "deg")
+        bird.roll = self._arg_with_dim(roll, "deg")
+        bird.insert(self._parent)
+        self.multiparts.append(bird)
+        return bird
+
+    @aedt_exception_handler
+    def add_environment(self, env_folder, global_offset=[0, 0, 0], yaw=0, pitch=0, roll=0):
+        """Add an Environment Multipart Component from Json file.
+
+         .. code-block:: json
+
+            {
+                "name": "open1",
+                "version": 1,
+                "class":"environment",
+                "xlim":["-5","95"],
+                "ylim":["-60","60"],
+                "parts": {
+                    "open_area": {
+                        "comp_name": "open1.a3dcomp",
+                        "offset":null,
+                        "rotation_cs":null,
+                        "rotation":null,
+                        "rotation_axis":null,
+                        "duplicate_number":null,
+                        "duplicate_vector":null
+                        }
+                }
+            }
+
+        Parameters
+        ----------
+        env_folder: str
+            Path to the actor folder. It must contain a json settings file and a 3dcomponent (.a3dcomp).
+        global_offset: list, Optional
+            Offset from Global Coordinate System.
+        yaw: float, Optional
+            Yaw Rotation from Global Coordinate System.
+        pitch: float, Optional
+            Pitch Rotation from Global Coordinate System.
+        roll: float, Optional
+            Roll Rotation from Global Coordinate System.
```

Comment:
```suggestion
        env_folder : str
            Path to the actor directory. It must contain a json settings file and a 
            3dcomponent (``.a3dcomp`` file).
        global_offset: list, optional
            Offset from global coordinate system.
        yaw : float, optional
            Yaw rotation from global coordinate system.
        pitch: float, optional
            Pitch rotation from global coordinate system.
        roll : float, optional
            Roll rotation from global coordinate system.
```

Reply:
Also, note that we should not be using "folder" in our variables.  That's a window GUI concept that has no meaning in linux batch world.  Directory is the correct term to use.